commentwriter.go

```diff
@@ -0,0 +1,42 @@
+package vfsgen
+
+import "io"
+
+// commentWriter writes given input to underlying io.Writer as a Go comment,
+// using line comments (//).
+type commentWriter struct {
+	io.Writer
+
+	wroteComment bool
+}
+
+func (cw *commentWriter) Write(p []byte) (n int, err error) {
+	for i, c := range p {
+		if c == '\n' {
+			if !cw.wroteComment {
+				_, err := cw.Writer.Write([]byte("//"))
+				if err != nil {
+					return n, err
+				}
+			}
+			cw.wroteComment = false
+		} else {
+			if !cw.wroteComment {
+				_, err := cw.Writer.Write([]byte("// "))
+				if err != nil {
+					return n, err
+				}
+				cw.wroteComment = true
+			}
+		}
+		_, err := cw.Writer.Write(p[i : i+1])
+		if err != nil {
+			return n, err
+		}
+		n++
+	}
+	if n != len(p) {
+		return n, io.ErrShortWrite
+	}
```

Comment:
If I'm right, these 3 lines are basically equivalent to:

```Go
if false {
	return n, io.ErrShortWrite
}
```

Can anyone confirm that's the case?